course_discovery/apps/edly/api/v1/views.py

```diff
@@ -0,0 +1,104 @@
+"""
+Views for API v1.
+"""
+from django.utils.text import slugify
+from rest_framework import (
+    status,
+    viewsets
+)
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from course_discovery.apps.edly.api.v1.serializers import EdlyProgramSerializer
+from course_discovery.apps.course_metadata.models import (
+    Course,
+    CourseRun,
+    Organization,
+    Program
+)
+
+
+class ProgramViewSet(viewsets.ModelViewSet):
+    """
+    Programs Create & Update.
+    """
+    serializer_class = EdlyProgramSerializer
+    permission_classes = (IsAuthenticated,)
+
+    def prepare_and_set_read_only_data(self, data, context):
+        """
+        Extracts read only data from data dictionary and sets it in context.
+        """
+        if data.get('org'):
+            organization = Organization.objects.filter(key=data['org']).first()
+            context['authoring_organization'] = organization
+            context['credit_backing_organization'] = organization
+
+        if data.get('course_runs'):
+            courses = []
+            excluded_course_runs = []
+            for course_run_key in data['course_runs']:
+                course_run = CourseRun.objects.filter(key=course_run_key).first()
```

Comment:
can there be multiple course_runs against a key? if not then this can be done without a loop by doing something like:

```
course_runs = CourseRun.objects.filter(key__in=data.get('course_runs'))
course_run_courses = course_runs.values('courses')
courses.extend(course_run_courses)
```

@zubair-arbi what do you think?

Reply:
Fancy 👍